src/main/java/edu/cornell/kfs/sys/dataaccess/impl/PreferencesDaoDynamoDB.java

```diff
@@ -193,11 +195,21 @@ public void saveUserPreferences(String principalName, Map<String, Object> prefer
 
     protected Table retrieveTable(String tableName) {
         DynamoDB dynamoDB = new DynamoDB(documentStoreClient.getClient());
+        if (StringUtils.isNotBlank(tableNamePrefix)) {
+            tableName = tableNamePrefix + tableName;
```

Comment:
If tableName is null, will this end up being prefix + "null"? Will dynamoDB.getTable() be able to handle the name accordingly?